sources/packages/backend/libs/integrations/src/institution-integration/ece-integration/ece-response.processing.service.ts

```diff
@@ -18,38 +45,285 @@ export class ECEResponseProcessingService {
+      );
+      processSummary.summary.push(
+        `Disbursements failed to process: ${disbursementsFailedToProcess}`,
+      );
+      this.logger.log(`SCompleted processing the file ${remoteFilePath}.`);
+    } catch (error: unknown) {
+      this.logger.error(error);
+      processSummary.errors.push(
+        `Error processing the file ${remoteFilePath}. ${error}`,
+      );
+    } finally {
+      // Delete the ECE response file, if the file exist in remote server.
+      if (isECEResponseFileExist) {
+        await this.deleteTheProcessedFile(remoteFilePath, processSummary);
+      }
+    }
+    return processSummary;
+  }
+
+  /**
+   * Sanitize the the detail records and transform the
+   * detail records to individual disbursements.
+   * @param eceFileDetailRecords detail records of ece file.
+   * @returns disbursements to be processed.
+   */
+  private sanitizeAndTransformToDisbursements(
+    eceFileDetailRecords: ECEResponseFileDetail[],
+  ): ECEDisbursements {
+    const disbursements = {} as ECEDisbursements;
+    for (const eceDetailRecord of eceFileDetailRecords) {
+      const errorMessage = eceDetailRecord.getInvalidDataMessage();
+      if (errorMessage) {
+        throw new Error(
+          `${errorMessage} at line ${eceDetailRecord.lineNumber}`,
+        );
+      }
+      const disbursement =
+        disbursements[eceDetailRecord.disbursementIdentifier];
+      if (!disbursement) {
+        disbursements[eceDetailRecord.disbursementIdentifier] = {
+          institutionCode: eceDetailRecord.institutionCode,
+          applicationNumber: eceDetailRecord.applicationNumber,
+          [AWARD_DETAILS_KEY]: [
+            {
+              payToSchoolAmount: eceDetailRecord.payToSchoolAmount,
+              isEnrolmentConfirmed: eceDetailRecord.isEnrolmentConfirmed,
+              enrolmentConfirmationDate:
+                eceDetailRecord.enrolmentConfirmationDate,
+            },
+          ],
+        };
+      } else {
+        const awardDetails =
+          disbursements[eceDetailRecord.disbursementIdentifier][
+            AWARD_DETAILS_KEY
+          ];
+        awardDetails.push({
+          payToSchoolAmount: eceDetailRecord.payToSchoolAmount,
+          isEnrolmentConfirmed: eceDetailRecord.isEnrolmentConfirmed,
+          enrolmentConfirmationDate: eceDetailRecord.enrolmentConfirmationDate,
+        });
+      }
+    }
+    return disbursements;
+  }
+
+  /**
+   * Validate the disbursement and application and
+   * update the enrolment status.
+   * @param disbursements disbursements to be processed,
+   * @param auditUser user who confirm or decline the enrolment.
+   * @param processSummary process summary.
+   * @returns count of disbursements processed, successfully updated, skipped and failed.
+   */
+  private async validateAndUpdateEnrolmentStatus(
+    disbursements: ECEDisbursements,
+    auditUserId: number,
+    processSummary: ProcessSummaryResult,
+  ): Promise<{
```

Comment:
If this is moved to a `class` we can also have the lines 218-221 initialized internally in the class.